src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/SentenceCalculation.kt

```diff
@@ -173,16 +175,46 @@ data class SentenceCalculation(
   }
 
   val earlyReleaseSchemeEligibilityDate: LocalDate? get() {
-    if (numberOfDaysToErsed == null) {
-      return null
+    if (calculateErsed) {
+      if (sentence.identificationTrack.calculateErsedFromHalfway()) {
+        return calculateErsedFromHalfway()
+      }
+      if (sentence.identificationTrack.calculateErsedFromTwoThirds()) {
+        return calculateErsedFromTwoThirds()
+      }
     }
-    return sentence.sentencedAt
-      .plusDays(numberOfDaysToErsed.toLong())
-      .plusDays(calculatedTotalAddedDays.toLong())
-      .minusDays(calculatedTotalDeductedDays.toLong())
-      .plusDays(calculatedTotalAwardedDays.toLong())
+    return null
   }
 
+  private fun calculateErsedFromTwoThirds(): LocalDate {
+    val custodialDuration = sentence.custodialDuration()
+    val days = custodialDuration.getLengthInDays(sentence.sentencedAt)
+    return if (days >= 1097) {
+      val release = if (extendedDeterminateParoleEligibilityDate != null) extendedDeterminateParoleEligibilityDate!! else releaseDate
+      release.minusYears(1)
+    } else {
+      sentence.sentencedAt
+        .plusDays(ceil(days.toDouble() / 3).toLong())
+        .plusDays(calculatedTotalAddedDays.toLong())
+        .minusDays(calculatedTotalDeductedDays.toLong())
+        .plusDays(calculatedTotalAwardedDays.toLong())
+    }
+  }
+
+  private fun calculateErsedFromHalfway(): LocalDate {
+    val custodialDuration = sentence.custodialDuration()
+    val days = custodialDuration.getLengthInDays(sentence.sentencedAt)
+    return if (days >= 1463) {
```

Comment:
that's an odd rule > 1463 days! can we stick the 1463 into a  named constant, presumably it represents some policy value?

Reply:
Done